lib/publiccloud/provider.pm

```diff
@@ -318,7 +318,12 @@ sub terraform_prepare_env {
         assert_script_run('curl ' . data_url("publiccloud/terraform/sap/$file.tfvars") . ' -o ' . TERRAFORM_DIR . "/$cloud_name/terraform.tfvars");
     }
     else {
-        assert_script_run('curl ' . data_url("publiccloud/terraform/$file.tf") . ' -o ' . TERRAFORM_DIR . '/plan.tf');
+        if (get_var('PUBLIC_CLOUD_TERRAFORM_FILE')) {
```

Comment:
I am not sure that we actually need this flexibility , usually we exposing something into openQA variable to have ability to clone job with different value for this variable and have different behavior at the end . I don't such possibility for this certain case . If you will try clone Azure NFS test with "normal " `azure.tf` it will not gonna work , right ? Also you can not set here any random file because it should be merged into `data` folder . 
I would recommend to keep "hardcoded" approach as it was before . So this whole block will be : 
```
my $tf_file = "publiccloud/terraform/" . (get_var('PUBLIC_CLOUD_AZURE_NFS_TEST') ? 'azure_nfstest.tf' : "$file.tf" );

assert_script_run('curl ' . data_url($tf_file) . ' -o ' . TERRAFORM_DIR . '/plan.tf');
```

Reply:
I believe the ability to inject a custom terraform profile is something that we anyway want to have in the long run. So instead of every time adding another hardcoded conditional, why not provide a custom terraform variable and hit this problem once and forever?

In general I appreciate the simple solution over the more complicated one, but I think having the ability to inject a custom terraform profile into our workflow is more than motivated to have in the longer run. Especially because now we have a use case for using a "non-default" terraform profile.

I believe in the long run it will be simpler to give the flexibility to choose a non-default profile at the level of openQA variables, instead of adding additional conditionals there every time we need to. It might be slimmer now, but then you add another, and another, and another, .... If you need to add a conditional once and you cannot exclude that more will come, then I prefer a strategy that allows us to avoid adding hardcoded conditionals at all.

WDYT?

Reply:
looks like you miss my main point . If we will merge your PR as is right now , we don't have any flexibility. You won't be able to run any tests except Azure NFS with profile which you have created and you won't be able to run Azure NFS test with any other profile than you have created .

Reply:
For this test case perhaps not, but for any future test runs, which require a different terraform profile.
Sure, we cannot predict if we ever need this flexibility. But I know that we need to add the ability to run a different terraform profile now and I don't see any good reason, why we should not go half a step further and give ourselves the ability to run any given terraform profile that we want.

While it does not allow us to run any terraform profile, it gives us the chance to run any **compatible** terraform profile (i.e. profiles that work with our test cases). If we then need to add another terraform profile, there is a high chance we don't need to add another conditional to this code segment but can just configure it via the introduced variable.

***

TLDR: This flexibility is cheap and hard-coded conditionals don't give us any benefit either. I don't see why this flexibility should hurt us, while I oppose hard-coded conditionals unless justified by simplicity or by necessity. Those two boundary conditions are IMHO not fulfilled here, as this PR proofs.

WDYT?

Reply:
NOTE: We getting into "programming style" kind of discussion , where is no correct answer but multiply and you can find supporters for each. So I want underline that it is just my personal opinion and I accepting the fact there might be different.

> For this test case perhaps not, but for any future test runs, which require a different terraform profile.

I always against such approach . The functionality which might be used at some point of time should be implemented at this point of time not earlier. Since my statement is fully abstract it is hard to prove it or deny it :). But I can just tell you that I have seen too many  cases where such code hit me back ...

> This flexibility is cheap 

I disagree with this statement from 2 different perspectives : 
1. Most probably you taking into account only "implementation" effort , but you also need to sum up time spent by each engineer who will need to trigger Public Cloud test and spent some extra time looking into code/documentation for yet-another-variable , and plus time when bot-ng/openQA/qac-openqa-yaml will mess up this variable with whatever random reason and someone will need to figure why test is failing and I can ensure you that investigation will be not obvious . And just to remind you as for now this price is for NOTHING as we don't have anything we can take back for it 
2. "Everything is measured in comparison ". You saying that flexibility is cheap and I am saying that it is cheap as effort needed to add same flexibility later when it will actually have a valid use case

Reply:
I was referring to THIS particular flexibility and here my argument still holds :slightly_smiling_face: 

The effort you are referring needs to be invested in all cases, so this point is not valid. Weather you would add an additional conditional or you would simply set the `PUBLIC_CLOUD_TERRAFORM_FILE` variable, any engineer would still need to invest time into openQA-specifics. At no point there was any statement that this would not be the case.

But: Changing a variable is relatively simple, adding a conditional with corresponding VRs and review is expensive (see this thread).

Ad 2) I never made a generalization and absolutely agree with your previous statement, that we should not introduce too much flexibility in places where it is probably never being used. So please also see my code suggestion as a specific case.

Reply:
Issue resolved in discussion, closing thread to avoid unnecessary wall-of-text